src/extensions/agent_modules/cloud_communication/adps2_mqtt_client/src/adps2_mqtt_client_module.c

```diff
@@ -201,17 +290,17 @@ bool ProcessDeviceRegistrationResponse(const char* payload, size_t payload_len)
     if (strncmp(status, "assigned", 8) == 0)
     {
         Log_Info("Device is registered.");
-        SetRegisterState(ADPS_REGISTER_STATE_REGISTERED, "received assigned status");
+        SetRegisterState(moduleState, ADPS_REGISTER_STATE_REGISTERED, "received assigned status");
```

Comment:
would it be better to have a an MqttStateStore object to fetch the state from so that do not need to plumb this moduleState or have it be part of interface object?  Here is example commit on experimental branch of store implementation backed by c++ impl: https://github.com/Azure/iot-hub-device-update/commit/10450fa57d6d2d124f9f261a42e2dea8391b285d